source/retryable-reads/tests/mapReduce-serverErrors.yml

```diff
@@ -0,0 +1,146 @@
+database_name: &database_name "retryable-reads-tests"
+collection_name: &collection_name "coll"
+
+data:
+    - {_id: 1, x: 11}
+    - {_id: 2, x: 22}
+    - {_id: 3, x: 33}
+    
+minServerVersion: '4.1.6' # 4.2
+
+tests:
+    -
+        description: "MapReduce succeeds after InterruptedAtShutdown"
+        clientOptions:
+            retryReads: true
+        failPoint: &failCommand_failPoint
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data: { failCommands: [mapReduce], errorCode: 11600 }
+        operation: &retryable_operation
+            name: mapReduce
+            object: collection
+            arguments:
+                map: "function inc() { return emit(0, this.x + 1) }"
+                reduce: "function sum(key, values) { return values.reduce((acc, x) => acc +x); }"
```

Comment:
These would be the first tests we have for `mapReduce` operations, but I think you should use Extended JSON for JavascriptWithoutScope BSON types here. I'm not sure if all drivers allow strings for these arguments, but all should accept Javascript BSON types.

Reply:
Also, spacing looks odd on `acc +x`. Should there be another space between `+` and `x`?

I also didn't realize the server supported `=>` syntax for its Javascript (ECMAscript 6 syntax?).